File: Codeforces/C++/Beautiful_Matrix.cpp
#include <iostream>
using namespace std;

int main() {
    int matrix[5][5];
    int p = 0, q = 0;

    for (int i = 0; i < 5; i++) {
        for (int j = 0; j < 5; j++) {
            cin >> matrix[i][j];
            if (matrix[i][j] == 1) {
                p = i;
                q = j;
            }
        }
    }

    p = (p < 2) ? (2 - p) : (p - 2);
    q = (q < 2) ? (2 - q) : (q - 2);

    cout << p+q <<endl;

    return 0;
}

File: Codeforces/Java/src/Beautiful_Matrix.java
import java.util.Scanner;

public class Beautiful_Matrix {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int p=0,q=0,x=0,y=0;
        int [][] matrix = new int[5][5];
        for (int i=0; i<5; i++){
            for(int j=0; j<5; j++){
                matrix[i][j]= sc.nextInt();
                if(matrix[i][j]==1){
                    p=i;
                    q=j;
                }
            }
        }
        if(p<2){
            p = 2-p;
        }
        else{
            p = p-2;
        }
        if(q<2){
            q = 2-q;
        }
        else{
            q = q-2;
        }
        System.out.println(p+q);
    }
}
